applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -617,6 +625,16 @@ static bool handle_hid_report_provider_event(struct hid_report_provider_event *e
 	__ASSERT_NO_MSG(!event->hid_state_api);
 	event->hid_state_api = &hid_state_api;
 
+	if (state.active_subscriber) {
+		struct report_state *rs = get_report_state(state.active_subscriber,
+							   event->report_id);
+		__ASSERT_NO_MSG(rs);
+
+		if (rs->provider == provider) {
+			link_provider_to_rs(provider, rs);
```

Comment:
Are we sure that after this call, `handle_hid_report_provider_event` will travel back to provider before it starts handling button event? If not, the provider might try to call non-existant handler (it will fail at your new assert). I guess it's workqueue so we should be fine.